Xi/listdev.c

```diff
@@ -376,22 +382,10 @@ ProcXListInputDevices(ClientPtr client)
         ListDeviceInfo(client, d, dev++, &devbuf, &classbuf, &namebuf);
     }
 
-    xListInputDevicesReply rep = {
-        .repType = X_Reply,
+    xListInputDevicesReply reply = {
         .RepType = X_ListInputDevices,
-        .sequenceNumber = client->sequence,
         .ndevices = numdevs,
-        .length = bytes_to_int32(total_length),
     };
 
-    if (client->swapped) {
-        swaps(&rep.sequenceNumber);
-        swapl(&rep.length);
-    }
-
-    WriteToClient(client, sizeof(xListInputDevicesReply), &rep);
-    WriteToClient(client, total_length, savbuf);
-    free(savbuf);
-    free(skip);
```

Comment:
We should keep this `free` call.

Reply:
indeed. we later can refactor this so that allocated chunk isn't needed anymore.